### PR TITLE
add logits option to generate artifacts

### PR DESCRIPTION
**Pull request by AdamLouly**
### Description

Adding the ability to export logits as an output for train and eval graphs in generate_artifacts
it will remain optional..

### CODE REVIEW

File: orttraining/orttraining/python/training/artifacts.py [hunk 1]
```diff
@@ -104,6 +105,13 @@ def __init__(self, _loss):
             self._loss = _loss
 
         def build(self, *inputs_to_loss):
+            if "additional_output_names" in extra_options:
+                loss_output = self._loss(*inputs_to_loss)
+                if isinstance(loss_output, tuple):
+                    return (*loss_output, *tuple(extra_options["additional_output_names"]))
```
Comment (pengwa) [inline orttraining/orttraining/python/training/artifacts.py:111]:
```suggestion
                    return (loss_output, *tuple(extra_options["additional_output_names"]))
```

File: orttraining/orttraining/test/python/orttraining_test_onnxblock.py [hunk 2]
```diff
@@ -847,6 +847,39 @@ def mse_loss(prediction, target):
...
+
+        # Make sure only loss is the output
+        assert len(eval_model.graph.output) == 1
+
+        # Re-generate artifacts with additional output names
+        artifacts.generate_artifacts(
+            onnx_model,
+            loss=artifacts.LossType.CrossEntropyLoss,
+            artifact_directory=temp_dir,
+            additional_output_names=["output-0"],
```
Comment (pengwa) [inline orttraining/orttraining/test/python/orttraining_test_onnxblock.py:875]: nit： do we need check the name existence in the graph output?

**Review by pengwa — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Comment by AdamLouly** [comment]
Thanks for the review
